src/packet/ipv6.rs

```diff
@@ -32,6 +32,86 @@ pub struct Ipv6 {
     payload: Vec<u8>,
 }
 
+/// Represents an IPv6 Packet pseudo header
+#[packet] pub struct Ipv6PseudoHeader {
+    #[construct_with(u16, u16, u16, u16, u16, u16, u16, u16)]
+    source: Ipv6Addr,
+    // RFC 2460:
+    // If the IPv6 packet contains a Routing header, the Destination Address used in the
+    // pseudo-header is that of the final destination. At the originating node, that address will
+    // be in the last element of the Routing header; at the recipient(s), that address will be in
+    // the Destination Address field of the IPv6 header.
```

Comment:
These comments should probably be turned into doc comments - the compiler will likely complain once you've fixed the other compile errors.